walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,89 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
-}
+
+    /** Check if the event and the request contains the same attendee. */
+    private boolean containSameAttendees(Collection < String > eventAttendees, Collection < String > requestAttendees) {
+        if (requestAttendees.size() == 0)
+            return false;
+
+        for (String attendee: eventAttendees)
+            if (requestAttendees.contains(attendee))
+                return true;
+        return false;
+    }
+
+    /**
+     * Combine overlapped TimeRanges in a sorted List, return collection without overlap
+     */
+    private Collection < TimeRange > combineTimeRanges(List < TimeRange > timeRanges) {
+        Collection < TimeRange > combinedTimeRanges = new ArrayList < > ();
+
+        if (timeRanges.size() == 0)
+            return combinedTimeRanges;
+
+        TimeRange currentEvent = timeRanges.get(0);
+        for (int i = 1; i < timeRanges.size(); i++) {
+            TimeRange nextEvent = timeRanges.get(i);
+            if (currentEvent.overlaps(nextEvent)) {
+                int start = currentEvent.start();
+                int end = currentEvent.contains(nextEvent.end()) ? currentEvent.end() : nextEvent.end();
```

Comment:
This should work, but using end = max(currentEvent.end(), nextEvent.end()) would be more succinct